### PR TITLE
Implement simplified surfaces and faster contouring

### PR DESCRIPTION
**Pull request by bclyons12**
This creates a simplified, lighter-weight version of flux surfaces suited to some applications (e.g., FRESCO). It also uses the new IMASutils contouring where applicable.

### CODE REVIEW

**Comment by bclyons12** [comment]
@orso82 I didn't modify `flux_surface()` since that has lots of fancy logic for open surfaces that I can't guarantee would be covered by IMASutil's contouring. It looks like we could do the same thing for `trace_surfaces()` that I did for `trace_simple_surfaces()`, and for the other version of `find_psi_boundary()` like I did for my version.

**Comment by bclyons12** [comment]
Also, once this PR is deemed ready to merge, this is a circumstance where we should run all the FUSE tests on it first. That will stress test the new contouring routines.

**Comment by orso82** [comment]
To run the FUSE tests using your changes, just open a PR in FUSE on a branch with the same `simple_surface` name that you have used here.

**Comment by bclyons12** [comment]
The IMAS 2.0 refactoring has been merged into this, which means both versions of `find_psi_boundary()` now use the new contouring.

**Comment by bclyons12** [comment]
@orso82 `trace_surfaces()` now uses the new IMASutils contouring as well. Only `flux_surface` (and other fucntions that depend on that) is left untouched because of the other types of surfaces that it can contour.

**Comment by bclyons12** [comment]
@orso82 All the FUSE regressions pass (https://github.com/ProjectTorreyPines/FUSE.jl/pull/770) so I'll merge this unless you want to make the `find_psi_boundary` more robust under this PR.

**Comment by orso82** [comment]
@bclyons12 this is great, although it's unclear to me what's the difference between `trace_surfaces` and `trace_simple_surfaces`.

Do we really need both?

If we can't just have a single version, could you add a description to the docstring of`trace_simple_surfaces` and perhaps highlight the difference with respect to `trace_surfaces`

**Comment by bclyons12** [comment]
@orso82 I cleaned up `trace_simple_surfaces` even more and added some documentation. The idea is that the FluxSurface struct has a lot of information, but if all we want is to perform flux surface averaging, we can get away with a lot less. In FRESCO where I might need to trace surfaces every iteration, this reduces the computation needed.

File: src/physics/fluxsurfaces.jl [hunk 1]
```diff
@@ -1613,18 +1613,24 @@ function flux_surfaces(eqt::equilibrium__time_slice{T1}, wall_r::AbstractVector{
...
-    npts_in_wall = sum(pts_in_wall)
-    if npts_in_wall == 0
-        error("Could not find guess for magnetic axis inside wall")
-    elseif npts_in_wall > 1
-        error("Multiple guesses for magnetic axis found inside wall")
-    end
-    ig, jg = pts[argmax(pts_in_wall)] # this are the indices for the guess inside the wall
-    rguess, zguess = r[ig], z[jg]
-    RA, ZA = find_magnetic_axis(r, z, PSI_interpolant, psi_sign; rguess, zguess)
+    if !isempty(wall_r)
```
Comment (orso82) [inline src/physics/fluxsurfaces.jl:1616]: Thank you for doing this @bclyons12 ! 👍 

I wonder if we should add this logic to the `find_magnetic_axis()` function.

Comment (bclyons12) [reply src/physics/fluxsurfaces.jl:1616]: You would have to pass the wall to it then, so up to you. You could make a separate function that takes the wall as an argument and does this I suppose.

**Comment by bclyons12** [comment]
@orso82 If the FUSE tests pass (https://github.com/ProjectTorreyPines/FUSE.jl/pull/770), I would like to merge this.

**Comment by bclyons12** [comment]
All FUSE tests pass (https://github.com/ProjectTorreyPines/FUSE.jl/pull/770)

**Pull request merged**